packages/shared/lib/services/sync/data/data.service.ts

```diff
@@ -35,13 +35,18 @@ export async function upsert(
     try {
         const encryptedRecords = encryptionManager.encryptDataRecords(recordsWithoutDuplicates);
 
-        await schema().from(RECORDS_TABLE).insert(encryptedRecords).onConflict(['nango_connection_id', 'external_id', 'model']).merge();
+        const externalIds = await schema()
+            .from<DataRecord>(RECORDS_TABLE)
+            .insert(encryptedRecords)
+            .onConflict(['nango_connection_id', 'external_id', 'model'])
+            .merge()
+            .returning('external_id');
 
         if (softDelete) {
             return {
                 success: true,
                 summary: {
-                    deletedKeys: [...addedKeys, ...updatedKeys],
```

Comment:
addedKeys would be empty, since it is checking if the ids is already in the db. UpdatedKeys is also empty if the record hash hasn't changed